lib/Models/ArcGisFeatureServerCatalogItem.js

```diff
@@ -146,13 +212,68 @@ ArcGisFeatureServerCatalogItem.prototype.showOnSeparateMap = function(
   }
 };
 
+function setupUniqueValRenderer(renderer) {
+  const out = {};
+  for (var i = 0; i < renderer.uniqueValueInfos.length; i++) {
+    const val = renderer.uniqueValueInfos[i].value;
+    out[val] = renderer.uniqueValueInfos[i];
+  }
+  return out;
+}
+
+function convertEsriRendererToStyle(symbol) {
```

Comment:
BTW I partially implemented this for the NSW Digital Twin:
https://github.com/TerriaJS/nsw-digital-twin/blob/master/lib/Models/EsriRestVectorCatalogItem.js#L145
Just in case it's helpful.

Reply:
Think we've both covered off different parts - I'll steal some of your bits to flesh mine out a bit further.